### ISSUE TITLE
[BUG] Ubuntu 20.04 Focal Fossa install

### ISSUE THREAD

**Issue by s41m0n** [polycube-network/polycube#297]
## Describe the bug
In the new Ubuntu 20.04 release they have updated rules concerning repositories and security. Thus, running the installation script lead to an error of the `go-lang` repository as follow:

```bash
johndoe@johndoe-vm:~/Desktop/polycube$ ./scripts/install.sh 
Use 'install.sh -h' to show advanced installation options.
+ set -e
+++ dirname ./scripts/install.sh
++ cd ./scripts
++ pwd
+ DIR=/home/johndoe/Desktop/polycube/scripts
+ source scripts/pre-requirements.sh
++ '[' -z /home/johndoe/Desktop/polycube/scripts ']'
++ mkdir -p /home/johndoe/dev
++ sudo apt update
[sudo] password for johndoe: 
Ign:1 http://ppa.launchpad.net/longsleep/golang-backports/ubuntu focal InRelease
Hit:2 http://security.ubuntu.com/ubuntu focal-security InRelease
Hit:3 http://it.archive.ubuntu.com/ubuntu focal InRelease                
Hit:4 http://it.archive.ubuntu.com/ubuntu focal-updates InRelease        
Err:5 http://ppa.launchpad.net/longsleep/golang-backports/ubuntu focal Release
  404  Not Found [IP: 91.189.95.83 80]
Hit:6 http://it.archive.ubuntu.com/ubuntu focal-backports InRelease
Reading package lists... Done
E: The repository 'http://ppa.launchpad.net/longsleep/golang-backports/ubuntu focal Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
+++ error_message
+++ set +x

Error during installation
***********************SYSTEM INFO*************************************
kernel version: 5.4.0-29-generic
linux release info: DISTRIB_ID=Ubuntu DISTRIB_RELEASE=20.04 DISTRIB_CODENAME=focal DISTRIB_DESCRIPTION="Ubuntu 20.04 LTS" NAME="Ubuntu" VERSION="20.04 LTS (Focal Fossa)" ID=ubuntu ID_LIKE=debian PRETTY_NAME="Ubuntu 20.04 LTS" VERSION_ID="20.04" HOME_URL="https://www.ubuntu.com/" SUPPORT_URL="https://help.ubuntu.com/" BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/" PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy" VERSION_CODENAME=focal UBUNTU_CODENAME=focal
g++ version: g++ (Ubuntu 9.3.0-10ubuntu2) 9.3.0 Copyright (C) 2019 Free Software Foundation, Inc. This is free software; see the source for copying conditions. There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
gcc version: gcc (Ubuntu 9.3.0-10ubuntu2) 9.3.0 Copyright (C) 2019 Free Software Foundation, Inc. This is free software; see the source for copying conditions. There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
./scripts/install.sh: line 11: go: command not found
golang-version:
```

## To Reproduce
Run the `install.sh`

## Expected behavior
Polycube installed.

## Please tell us about your environment: 
Fresh installation of Ubuntu 20.04
```bash
NAME="Ubuntu"
VERSION="20.04 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal

```

## Possible Solutions
1- In the `script/pre-requirements.sh` file the dependency `golang` could be modified to match this new OS version and installed via `apt install golang` (*TESTED AND WORKING*)
2 - In the `script/pre-requirements.sh` file the dependency `golang` could be modified to match this new OS version and installed via `snap install go`

I think that if a ppa does not have a Release file for a specific distribution could mean that:
* it does not support your os version (since you should install it differently)
* it is not updated

**Comment by s41m0n** [comment]
@frisso @sebymiano Please tell me whether this could be useful or not

**Comment by frisso** [comment]
Simone, I would go with default install option, so (1).
I would just suggest to add a test so that apt install is used only with ubuntu >= 20.04.
Finally, we should update the manual installation procedure (in readthedocs) as well.
Many thanks!

**Comment by s41m0n** [comment]
Solved with #299
